xds/src/test/java/com/linecorp/centraldogma/xds/internal/DiscoveryServiceItTest.java

```diff
@@ -0,0 +1,274 @@
+import com.linecorp.centraldogma.common.Change;
+import com.linecorp.centraldogma.server.CentralDogmaBuilder;
+import com.linecorp.centraldogma.testing.junit.CentralDogmaExtension;
+
+import io.envoyproxy.controlplane.cache.VersionedResource;
+import io.envoyproxy.envoy.config.cluster.v3.Cluster;
+import io.envoyproxy.envoy.config.cluster.v3.Cluster.DiscoveryType;
+import io.envoyproxy.envoy.config.core.v3.Address;
+import io.envoyproxy.envoy.config.core.v3.ApiConfigSource;
+import io.envoyproxy.envoy.config.core.v3.ApiVersion;
+import io.envoyproxy.envoy.config.core.v3.ConfigSource;
+import io.envoyproxy.envoy.config.core.v3.GrpcService;
+import io.envoyproxy.envoy.config.core.v3.GrpcService.EnvoyGrpc;
+import io.envoyproxy.envoy.config.core.v3.SocketAddress;
+import io.envoyproxy.envoy.config.core.v3.SocketAddress.Protocol;
+import io.envoyproxy.envoy.config.endpoint.v3.ClusterLoadAssignment;
+import io.envoyproxy.envoy.config.endpoint.v3.Endpoint;
+import io.envoyproxy.envoy.config.endpoint.v3.LbEndpoint;
+import io.envoyproxy.envoy.config.endpoint.v3.LocalityLbEndpoints;
+import io.envoyproxy.envoy.config.listener.v3.Filter;
+import io.envoyproxy.envoy.config.listener.v3.FilterChain;
+import io.envoyproxy.envoy.config.listener.v3.Listener;
+import io.envoyproxy.envoy.config.route.v3.Route;
+import io.envoyproxy.envoy.config.route.v3.RouteAction;
+import io.envoyproxy.envoy.config.route.v3.RouteConfiguration;
+import io.envoyproxy.envoy.config.route.v3.RouteMatch;
+import io.envoyproxy.envoy.config.route.v3.VirtualHost;
+import io.envoyproxy.envoy.extensions.filters.http.router.v3.Router;
+import io.envoyproxy.envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager;
+import io.envoyproxy.envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.CodecType;
+import io.envoyproxy.envoy.extensions.filters.network.http_connection_manager.v3.HttpFilter;
+import io.envoyproxy.envoy.extensions.filters.network.http_connection_manager.v3.Rds;
+
+@Testcontainers(disabledWithoutDocker = true)
+final class DiscoveryServiceItTest {
+
+    private static final String API_CONFIG_SOURCE_CLUSTER_NAME = "xds_cluster";
+
+    private static final String ECHO_CLUSTER = "echo_cluster";
+    private static final String ECHO_CLUSTER_ADDRESS = "echo_upstream";
+    private static final String NO_ECHO_CLUSTER = "no_echo_cluster";
+    private static final String NO_ECHO_CLUSTER_ADDRESS = "no_echo_upstream";
+
+    private static final String ECHO_ROUTE = "echo_route";
+    private static final String ECHO_LISTENER = "echo_listener";
+
+    // Using 10000 is fine because this port is used in the container. The port is exposed with a different
+    // port number which you can get via ContainerState.getMappedPort().
+    private static final Integer ENVOY_LISTENER_PORT = 10000;
+
+    @RegisterExtension
+    final CentralDogmaExtension dogma = new CentralDogmaExtension() {
```

Comment:
```suggestion
    static final CentralDogmaExtension dogma = new CentralDogmaExtension() {
```

Reply:
I believe that we should omit `static` when `runForEachTest` is `true`.

Reply:
I see, I guess we leave it as is then 😅